### PR TITLE
[UPLOAD-936] [UPLOAD-1005] [UPLOAD-1023] native oauth login

### PR DESCRIPTION
**Pull request by krystophv**
for [UPLOAD-936], provide the standard oauth login flow via default browser (subtasks [UPLOAD-973], [UPLOAD-982], and [UPLOAD-989]; [WEB-2144] [WEB-2150])
also [UPLOAD-934], a small formatting change

[UPLOAD-936]: https://tidepool.atlassian.net/browse/UPLOAD-936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UPLOAD-934]: https://tidepool.atlassian.net/browse/UPLOAD-934?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UPLOAD-973]: https://tidepool.atlassian.net/browse/UPLOAD-973?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UPLOAD-982]: https://tidepool.atlassian.net/browse/UPLOAD-982?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UPLOAD-989]: https://tidepool.atlassian.net/browse/UPLOAD-989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[WEB-2144]: https://tidepool.atlassian.net/browse/WEB-2144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-2150]: https://tidepool.atlassian.net/browse/WEB-2150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by clintonium-119 — APPROVED** [review]
Nice work :100: LGTM

**Review by clintonium-119 — DISMISSED** [review]
Nice work.  All updates LGTM :ship:

**Review by clintonium-119 — DISMISSED** [review]
LGTM :+1:

File: styles/components/LoggedOut.module.less [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/**
+* Copyright (c) 2014, Tidepool Project
```
Comment (tjotala) [inline styles/components/LoggedOut.module.less:2]: This appears to be a new file, but the year is 2014. Should be it 2023?

File: test/app/actions/async.test.js [hunk 2]
```diff
@@ -1403,26 +1405,21 @@ describe('Asynchronous Actions', () => {
       __Rewire__('services', {
         api: {
           user: {
-            logout: (cb) => cb(null)
+            logout: (cb) => cb('Error :(')
```
Comment (tjotala) [inline test/app/actions/async.test.js]: Do we actually show this? :)

**Review by tjotala — APPROVED** [review]
LGTM except for couple of questions

File: styles/components/LoggedOut.module.less (hunk 1 above)
Comment (krystophv) [reply styles/components/LoggedOut.module.less:2]: Ah, yeah, just used another file to spawn this one. We actually generally don't include the license and copyright block on new files. I can get rid of it.

File: test/app/actions/async.test.js (hunk 2 above)
Comment (krystophv) [reply test/app/actions/async.test.js]: No, purely a test fixture. 🙂

**Review by clintonium-119 — DISMISSED** [review]
LGTM :ship: 
This must have been a fun one to figure out!

**Review by clintonium-119 — APPROVED** [review]
LGTM :100:

**Pull request merged**